### DISCUSSION TITLE
Unable to login to custom app using azure

### DISCUSSION THREAD

**Discussion by BasBasHardbas** [rustdesk/rustdesk#8999] [Q&A] [answered]
Hello, I have just made custom client for our company and when I try to login in app there is no option to use azure.
Also the client seems not to apply strategies.
Override Settings is disabled as well as Default Settings.
I'm using M1 MacBook Pro

**Comment by rustdesk** [comment]
It seems API is not set correctly. You can try login with user/password, just type any name and password, and click ok, you will see if the API works.

![image](https://github.com/user-attachments/assets/37b61beb-d5ad-461c-85a2-0f42c2f870fe)

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/wiki/FAQ#failed-to-log-in-with-custom-client-build-or-public-release-with-wrong-api-settings:~:text=set%20up%20your-,https,-.

**Reply by BasBasHardbas** [reply]
It seems that removing the port helped with strategies, but I still cannot login using azure.

**Reply by rustdesk** [reply]
> You can try login with user/password, just type any name and password, and click ok, you will see if the API works.

Do this please?

**Reply by BasBasHardbas** [reply]
Done it, getting error as showed on the screen 
![image](https://github.com/user-attachments/assets/644d9193-0658-4767-821b-761a41a17cf2)

**Reply by rustdesk** [reply, marked as answer]
The error message has told you everything. It seems your api is not correct. Make sure the hidden uri part`http://example.com` in your picture can be open on your browser. You may have change to `https`, if so, please change current `http` to `https`.

> http://example.com

shoud be same to uri (removing `/api/login` part) which you use to open the web console.

**Reply by BasBasHardbas** [reply]
Setting api address the same as web console address fixed everything. Thank you for your time.
